services/watson-extension/src/watson_extension/clients/platform/sources.py

```diff
@@ -80,18 +83,38 @@ async def get_sources(
 
         return True, None
 
+    async def is_source_name_valid(self, integration_setup_name: str) -> bool:
+        request = "/api/sources/v3.1/graphql"
+        response = await self.platform_request.post(
+            self.sources_url,
+            request,
+            json={
+                "query": f'{{ sources(filter: {{name: "name", value: "{integration_setup_name}"}}){{ id, name }}}}'
+            },
+        )
+
+        if response.ok:
+            content = await response.json()
+            if len(content.get("data").get("sources")) == 0:
+                return True
+
+        return False
+
     async def bulk_create(
         self,
         integration_setup_name: str,
         integration_setup_redhat_cluster_identifier: str,
     ) -> bool:
         request = "/api/sources/v3.1/bulk_create"
+        validated_integration_setup_name = await self.is_source_name_valid(
+            integration_setup_name
+        )
         request_json = {
             "applications": [
                 {
                     "application_type_id": "2",
                     "extra": {"hcs": False},
-                    "source_name": integration_setup_name,
+                    "source_name": validated_integration_setup_name,
```

Comment:
@tahmidefaz could you double check this part?
It looks like `validated_integration_setup_name` is a bool ? did you mean to use `integration_setup_name` here and below (line 130) ?

Reply:
@josejulio Good one! I removed the call to `is_source_name_valid` from `bulk_create`. Let's bake in this logic in the Core.